app/backend/approaches/chatreadretrieveread.py

```diff
@@ -83,13 +83,33 @@ def __init__(
         self.query_speller = query_speller
         self.chatgpt_token_limit = get_token_limit(chatgpt_model)
 
+    @overload
```

Comment:
I haven't used overload before, how do the additional signatures aid type checkers? Were these added due to a mypy error?

Reply:
It was added because streaming=True returns a completely different type and instead of having Union's propagating from the return signature through the class and having big, messy type signatures this makes it simpler.